### ISSUE TITLE
VCSA crash (vpxd) when using AddStandaloneHost_Task() on rootFolder

### ISSUE THREAD

**Issue by dagwieers** [vmware/pyvmomi#554]
By accident I discovered that one can crash the vpxd service by doing this:

    task = content.rootFolder.AddStandaloneHost_Task(host_connect_spec, compute_resource_spec, as_connected, license)

Where:

    compute_resource_spec = None
    license = None

If you do this (correctly) on `datacenter.hostFolder` it works as expected.

So the fact that vpxd crashes when doing this seems to be a defect, and possibly with security implications.

**Comment by tianhao64** [comment]
@dagwieers Thanks for reporting this. 
I can't reproduce this on 6.5. Can you share more info?
What's your vCenter version? 
Can you share the server log related to the crushing?

**Comment by dagwieers** [comment]
This is on VMware vCenter Server Appliance 6.0.0.0 (6.0.0.0 Build 4811166).

**Comment by dagwieers** [comment]
Here is a minimal reproducer:
```python
#!/usr/bin/python

import ssl
from pyVim import connect
from pyVmomi import vim

# VCSA login information
vcsa_host = '172.16.1.123'
vcsa_user = 'administrator@vsphere.local'
vcsa_pass = 'yourpassword'

context = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
context.verify_mode = ssl.CERT_NONE
service_instance = connect.SmartConnect(host=vcsa_host, user=vcsa_user, pwd=vcsa_pass, sslContext=context)
content = service_instance.RetrieveContent()

# NOTE: This ESXi host does not have to exist to make it crash
host_connect_spec = vim.host.ConnectSpec()
host_connect_spec.hostName = '1.2.3.4'
host_connect_spec.userName = 'foo'
host_connect_spec.password = 'bar'

task = content.rootFolder.AddStandaloneHost_Task(host_connect_spec, None, True, None)

while True:
    if task.info.state == vim.TaskInfo.State.success:
        print task.info.result
        break
    elif task.info.state == vim.TaskInfo.State.error:
        print task.info.error
        break
    elif task.info.state in (vim.TaskInfo.State.running, vim.TaskInfo.State.queued):
        time.sleep(15)
```

**Comment by dagwieers** [comment]
I don't know what log file content you are looking for.
It seems the vxpd process did not crash, but it's hard to find out what exactly crashed.
After running this, it returns with:
```
Traceback (most recent call last):
  File "crash-vcsa.py", line 21, in <module>
    task = content.rootFolder.AddStandaloneHost_Task(host_connect_spec, None, True, None)
  File "/usr/lib/python2.7/site-packages/pyVmomi/VmomiSupport.py", line 580, in <lambda>
    self.f(*(self.args + (obj,) + args), **kwargs)
  File "/usr/lib/python2.7/site-packages/pyVmomi/VmomiSupport.py", line 386, in _InvokeMethod
    return self._stub.InvokeMethod(self, info, args)
  File "/usr/lib/python2.7/site-packages/pyVmomi/SoapAdapter.py", line 1324, in InvokeMethod
    resp = conn.getresponse()
  File "/usr/lib64/python2.7/httplib.py", line 1089, in getresponse
    response.begin()
  File "/usr/lib64/python2.7/httplib.py", line 444, in begin
    version, status, reason = self._read_status()
  File "/usr/lib64/python2.7/httplib.py", line 408, in _read_status
    raise BadStatusLine(line)
httplib.BadStatusLine: ''
```
And my vSphere Client automatically disconnects and I can no longer use the API (or pyvmomi). It will always report the above traceback.

The rhttpproxy log start to produce this:
```
2017-05-31T13:44:56.020Z warning rhttpproxy[7FB1385BF700] [Originator@6876 sub=Proxy Req 15197 Tunnel] Error on read from server: N7Vmacore15SystemExceptionE(Connection reset by peer)
2017-05-31T13:44:56.020Z warning rhttpproxy[7FB1385BF700] [Originator@6876 sub=Proxy Req 15200] Server localhost : 8085 disconnected with a response pending
2017-05-31T13:44:56.024Z warning rhttpproxy[7FB13EF46780] [Originator@6876 sub=Default] Failed to connect socket; <io_obj p:0x00007fb13f3227b0, h:16, <TCP '0.0.0.0:0'>, <TCP '127.0.0.1:8089'>>, e: system:111(Connection refused)
2017-05-31T13:44:56.024Z warning rhttpproxy[7FB13EF46780] [Originator@6876 sub=Proxy Req 15203 Tunnel] Error connecting tunnel on TCP socket: (null): Connection refused
2017-05-31T13:44:56.063Z warning rhttpproxy[7FB13EF46780] [Originator@6876 sub=Default] Failed to connect socket; <io_obj p:0x00007fb13f3227b0, h:16, <TCP '0.0.0.0:0'>, <TCP '127.0.0.1:8089'>>, e: system:111(Connection refused)
2017-05-31T13:44:56.063Z warning rhttpproxy[7FB13EF46780] [Originator@6876 sub=Proxy Req 15204 Tunnel] Error connecting tunnel on TCP socket: (null): Connection refused
```
And a few minutes later I think some watchdog process discovers the error and I get this on screen:

    The system is going DOWN for reboot in 5 minutes!

**Comment by prziborowski** [comment]
@dagwieers Thank you for reporting this issue.
It should be addressed in the next update/patch release of 6.0.

For the time being, try to avoid attempting to add a standalone host to the rootFolder. 😃

**Comment by dagwieers** [comment]
@prziborowski I'll try to resist the temptation.

Where do I send my account number for the bug bounty ? :-)

**Comment by DanielDraganov** [comment]
Fixed since vSphere 6.0 P06. Closing this issue.
